pyts/metrics/dtw.py

```diff
@@ -530,36 +609,35 @@ def itakura_parallelogram(n_timestamps, max_slope=2.):
      [1 3 4 4 5]]
 
     """
-    if not isinstance(n_timestamps, (int, np.integer)):
-        raise TypeError("'n_timestamps' must be an intger.")
-    else:
-        if not n_timestamps >= 2:
-            raise ValueError("'n_timestamps' must be an integer greater than "
-                             "or equal to 2.")
-    if not isinstance(max_slope, (int, np.integer, float, np.floating)):
-        raise TypeError("'max_slope' must be an integer or a float.")
-    else:
-        if not max_slope >= 1:
-            raise ValueError("'max_slope' must be a number greater "
-                             "than or equal to 1.")
+    if n_timestamps_2 is None:
+        n_timestamps_2 = n_timestamps_1
 
-    min_slope = 1 / max_slope
+    min_slope, max_slope = _get_itakura_slopes(n_timestamps_1, n_timestamps_2,
+                                               max_slope)
+    # Now we create the piecewise linear functions defining the parallelogram
+    # lower_bound[0] = min_slope * x
+    # lower_bound[1] = max_slope * (x - n_timestamps_1) + n_timestamps_2
+
+    centered_scale = np.arange(n_timestamps_1) - n_timestamps_1 + 1
+    lower_bound = np.empty((2, n_timestamps_1))
+    lower_bound[0] = min_slope * np.arange(n_timestamps_1)
+    lower_bound[1] = max_slope * centered_scale + n_timestamps_2 - 1
 
-    lower_bound = np.empty((2, n_timestamps))
-    lower_bound[0] = min_slope * np.arange(n_timestamps)
-    lower_bound[1] = ((1 - max_slope) * (n_timestamps - 1)
-                      + max_slope * np.arange(n_timestamps))
-    lower_bound = np.round(lower_bound, 2)
```

Comment:
Do you think that the rounding is no longer necessary? I encountered some float precision issues when I wrote this function the first time. For instance, if the theoretical value is 2 but the value is 1.9999999 or 2.000001, it messes up the results after flooring/ceiling. I don't think that it hurts much to keep it (rounding to 1 decimal is enough though).

Reply:
Good catch, you are actually right I noticed that too, I forgot to put them back :)